### PR TITLE
feat: Add csv as a valid input format for GCSTOJDBC Java template

### PR DESCRIPTION
**Pull request by nj1973**
Added CSV to valid input formats and added relevant code to read section of template.

### CODE REVIEW

File: java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCSToJDBC.java [hunk 1]
```diff
@@ -47,13 +54,39 @@ public static GCSToJDBC of(String... args) {
...
+                .option(GCS_JDBC_CSV_HEADER, true)
+                .option(GCS_JDBC_CSV_INFER_SCHEMA, true)
+                .load(config.getInputLocation());
+        break;
+      case GCS_JDBC_AVRO_FORMAT:
+      case GCS_JDBC_ORC_FORMAT:
+        inputData = spark.read().format(config.getInputFormat()).load(config.getInputLocation());
+        break;
+      case GCS_JDBC_PRQT_FORMAT:
+        inputData = spark.read().parquet(config.getInputLocation());
```
Comment (shashank-google) [inline java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCSToJDBC.java:80]: I would prefer if we can use same code as for Avro and ORC. 
Why are we using a different code for parquet ?
`inputData = spark.read().format(config.getInputFormat()).load(config.getInputLocation());`

File: java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCSToJDBC.java [hunk 2]
```diff
@@ -47,13 +54,39 @@ public static GCSToJDBC of(String... args) {
...
+    Dataset<Row> inputData = null;
+
+    switch (config.getInputFormat()) {
+      case GCS_JDBC_CSV_FORMAT:
+        inputData =
+            spark
+                .read()
+                .format(config.getInputFormat())
+                .option(GCS_JDBC_CSV_HEADER, true)
+                .option(GCS_JDBC_CSV_INFER_SCHEMA, true)
```
Comment (shashank-google) [inline java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCSToJDBC.java:72]: If we are now entertaining CSV as well, then we should also add "delimiter" as input too.
In addition there can be many more other configurable options like (dateFormat, nullValue, escape etc).
This is open for discussion.

Taking clues from respective Dataflow template.
https://cloud.google.com/dataflow/docs/guides/templates/provided-batch#gcs_text_to_cloud_spanner

**Review by shashank-google — CHANGES_REQUESTED** [review]

File: java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCSToJDBC.java (hunk 1 above)
Comment (nj1973) [reply java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCSToJDBC.java:80]: I followed what had been done previously in the `GCSToBigQuery.java` template. That one has hardcoded options for `inferSchema` and `header`. So does the equivalent Python `gcs_to_jsbc.py` template.

File: java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCSToJDBC.java (hunk 2 above)
Comment (nj1973) [reply java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCSToJDBC.java:72]: In keeping with trying to be consistent with other templates I only added the same two hardcoded options. I agree these these should be configurable and we should look to add more control. But I think that is something to be implemented across all CSV related templates and not just this one.
I believe it should be an issue in its own right.

Comment (nj1973) [reply java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCSToJDBC.java:72]: Agreed verbally that there should be another issue raised for full CSV option control across all relevant templates.

Comment (shashank-google) [reply java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCSToJDBC.java:72]: +1

**Review by shashank-google — APPROVED** [review]

**Pull request merged**